Update on WebView2 no-elevation installer and minor edits... (#1637)
MicrosoftDocs/edge-developer: microsoft-edge/webview2/concepts/distribution.md

Only one installation of the Evergreen WebView2 Runtime is needed for all Evergreen apps on the device.  Several tools are available at [Download the WebView2 Runtime](https://developer.microsoft.com/microsoft-edge/webview2#download-section) to help you deploy the Evergreen Runtime.

* For online clients: _WebView2 Runtime Bootstrapper_ is a tiny (approximately 2 MB) installer.  The WebView2 Runtime Bootstrapper downloads and installs the Evergreen Runtime from Microsoft servers that matches the user's device architecture.
  * In the setup part of your WebView2 app, link to the bootstrapper.  Use a link to programmatically download the bootstrapper; select the **Get the Link** button at the above download page.
  * Or, download the bootstrapper and package it with your WebView2 app.

* For offline clients: _WebView2 Runtime Standalone Installer_ is a full installer that installs the Evergreen WebView2 Runtime in offline environments.

### Installing the Runtime as per-machine or per-user

<!-- keep the 3 instances or variants of this passage in sync: -->

The latest bootstrapper and standalone installer support both _per-machine_ and _per-user_ installs of the WebView2 Runtime.

> If you run the installer from an elevated process or command prompt, the Runtime is installed as _per-machine_.  If you don't run the installer from an elevated process or command prompt, the Runtime will be installed as _per-user_.  However, a _per-user_ install is automatically replaced by a _per-machine_ install, if a _per-machine_ Microsoft Edge Updater is in place.  A _per-machine_ Microsoft Edge Updater is included as part of Microsoft Edge, except for the Canary preview channel of Microsoft Edge.

Reviewer: @liminzhu - Is Edge Updater always inherently per-machine, or are there two versions of it, a per-user version & a per-machine version?

Author: It depends on how the browser or WV is getting installed. If Edge is installed per machine (and for Stable, Beta, Dev, it is per-machine) then it's per machine. Canary is installed per user, so the updater is per user as well. What I mean to say in this paragraph is really if you have per-machine installed Edge (which should be the cases for all channels except Canary), the per machine updater that comes with it will put a per-machine WebVIew2 Runtime to replace the per-user one at some point.

Reviewer: @liminzhu - please review my added clarification here of the 1st instance of the paragraph.  In instance 2 & 3 of the paragraph, I added a link pointing to this section, to read the added detail.

Author: @mikehoffms thanks! I love having a `### Installing the Runtime as per-machine or per-user` that is referenced by other parts of the doc.

```
For the Stable channel of Microsoft Edge, the Beta preview channel of Microsoft Edge, and the Dev preview channel of Microsoft Edge, Microsoft Edge is installed per-machine, and so Microsoft Edge Updater is installed per-machine. If you have installed Microsoft Edge per-machine, the per-machine Microsoft Edge Updater that comes with Microsoft Edge installs a per-machine WebView2 Runtime to replace the per-user WebView2 Runtime at some point.

The Canary preview channel of Microsoft Edge is installed per-user, and so Microsoft Edge Updater is installed per-user.
```

This part does feel a bit repetitive as we're re-iterating what we were saying in the two sentences above that (`However, a _per-user_ install is automatically replaced by a _per-machine_ install, if a _per-machine_ Microsoft Edge Updater is in place.  A _per-machine_ Microsoft Edge Updater is included as part of Microsoft Edge, except for the Canary preview channel of Microsoft Edge.`). Replacing per-user with per-machine install is also not an important aspect that a developer must dwell on, and more of an optimization on our end, so I'd suggest keeping the two sentences and getting rid of the long version so that readers move on to other content faster.

Author: I have one more edit about regkey I want to squeeze in.

Author: @mikehoffms see the new commit i've made and let me know what you think about the above and whether we're ready to merge. Thanks!

Reviewer: will do

Reviewer: "keeping the two sentences and getting rid of the long version"
Removed the long version.

*   For offline clients: _WebView2 Runtime Standalone Installer_ is a full installer that installs the Evergreen WebView2 Runtime in offline environments.

### Installing the Runtime as per-machine or per-user

<!-- keep the 3 instances or variants of this passage in sync: -->

The latest bootstrapper and standalone installer support both _per-machine_ and _per-user_ installs of the WebView2 Runtime.

> If you run the installer from an elevated process or command prompt, the Runtime is installed as _per-machine_.  If you don't run the installer from an elevated process or command prompt, the Runtime will be installed as _per-user_.  However, a _per-user_ install is automatically replaced by a _per-machine_ install, if a _per-machine_<!--if updater is always per-machine, strike that "per-machine" qualifier, and break out a separate statement or phrase, that the updater is per-machine--> Microsoft Edge Updater is in place.  A _per-machine_<!--strike "A _per-machine_" if updater is always per-machine--> Microsoft Edge Updater is included as part of Microsoft Edge, except for the Canary preview channel of Microsoft Edge.

Reviewer: @liminzhu - In the 3 versions of this paragraph in this article, please remove the HTML comments `<!-- -->` I added.  Please clarify whether the updater is inherently per-machine, or if there are two versions of the updater: a per-machine updater and a per-user updater.

*   For offline clients: _WebView2 Runtime Standalone Installer_ is a full installer that installs the Evergreen WebView2 Runtime in offline environments.

> > [!NOTE]

Reviewer: global in article: check consistency of usage of phrases:
per-user
per-system
per-machine

Author: good catch, they are interchangeable and replaced all instances with per-machine

Reviewer: Resolution/findings:
* per-user - 3 hits
* per-machine - 13 hits, per-system - 0 hits

So our lexicon in this article is: Use "per-user" & "per-machine".  Write "per-machine" instead of "per-system".

> > The latest bootstrapper and standalone installer support both per-machine and per-user installs. If the installer is run with elevation, the Runtime will be installed per-system. Otherwise, the Runtime will be installed as per-user and and per-user install may be automatically replaced by a per-system install if a per-system Edge Updater (which should come with a non-Canary Edge browser) is in place.

Reviewer: ```suggestion
> The latest bootstrapper and standalone installer support both per-machine and per-user installs. If the installer is run with elevation, the Runtime will be installed as "per-system". Otherwise, the Runtime will be installed as "per-user".  Per-user install is automatically replaced by a per-system install if a per-system Edge Updater is in place.  A per-system Edge Updater is included as part of the Edge browser, except for Canary.
```

> 

Reviewer: Global: probably change "Edge Updater" to "Microsoft Edge Updater".  Full-text search in repo finds two hits on "Edge Updater":
* "Microsoft Edge pings the Microsoft Edge updater service about the progress of installation and update."
* "The recommended way of receiving WebView2 updates is by using the default Microsoft Edge updater."

        HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\EdgeUpdate\Clients\{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}
        ```

    *   Alternatively, run [GetAvailableCoreWebView2BrowserVersionString](/microsoft-edge/webview2/reference/win32/webview2-idl#getavailablecorewebview2browserversionstring) and evaluate whether the `versionInfo` is `nullptr`.  `nullptr` indicates that the Runtime isn't installed. Note that the API will return version info for non-Stable Edge channels as well if installed, and non-Stable Edge channels can be used for WebView2.

1.  If the Runtime is not installed, in your app setup process, use the link (from the **Get the Link** button on the download page) to programmatically download the WebView2 Runtime Bootstrapper.

> 1.  Invoke the WebView2 Runtime Bootstrapper by issuing the command `MicrosoftEdgeWebview2Setup.exe /silent /install`. Note that running the commmand from an elevated process or command prompt will trigger a per-system install, otherwise a per-machine install will take place and may be automatically replaced by a per-system install if a per-system Edge Updater (which should come with a non-Canary Edge browser) is in place.

Reviewer: ```suggestion
1.  Invoke the WebView2 Runtime Bootstrapper by issuing the command `MicrosoftEdgeWebview2Setup.exe /silent /install`. Running that command from an elevated process or command prompt will trigger a "per-system" install, otherwise a "per-machine" install will take place.  A "per-machine" install will be automatically replaced by a "per-system" install if a per-system Edge Updater is in place.  A per-system Edge Updater is provided as part of Microsoft Edge, except for Canary.
```
"per-user" install, or "per-machine" install?

        HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\EdgeUpdate\Clients\{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}
        ```

>     *   Alternatively, call [GetAvailableCoreWebView2BrowserVersionString](/microsoft-edge/webview2/reference/win32/webview2-idl#getavailablecorewebview2browserversionstring) and check whether the `versionInfo` is `nullptr`.  If `versionInfo` is `nullptr`, the WebView2 Runtime isn't currently installed on the client. Note that the API will return version info for non-Stable Edge channels as well if installed, and non-Stable Edge channels can be used for WebView2.

Reviewer: ```suggestion
    *   Alternatively, call [GetAvailableCoreWebView2BrowserVersionString](/microsoft-edge/webview2/reference/win32/webview2-idl#getavailablecorewebview2browserversionstring) and check whether the `versionInfo` is `nullptr`.  If `versionInfo` is `nullptr`, the WebView2 Runtime isn't currently installed on the client. If a preview channel of Microsoft Edge is installed, such as Beta, Dev, or Canary, the API returns version info for that preview channel.   Preview channels of Microsoft Edge can be used for WebView2.
```

> 1.  If the WebView2 Runtime is not installed, run the Evergreen Standalone Installer.  If you want to run a silent installation, you can run the command below. Note that running the commmand from an elevated process or command prompt will trigger a per-system install, otherwise a per-machine install will take place and may be automatically replaced by a per-system install if a per-system Edge Updater (which should come with a non-Canary Edge browser) is in place.

Reviewer: ```suggestion
1.  If the WebView2 Runtime is not installed, run the Evergreen Standalone Installer.  If you want to run a silent installation, you can run the command below.  Running the below command from an elevated process or command prompt will trigger a "per-system" install.  Otherwise, a "per-machine" install will take place.  A "per-machine" install is automatically replaced by a "per-system" install if a per-system Edge Updater is in place.  A per-system Edge Updater is provided with preview channels of Microsoft Edge.
```
... except for Canary?
"per-system" vs. "per-machine" vs "per-user"
